### ISSUE TITLE
Suggestion: Markdown for this component

### ISSUE THREAD

**Issue by irineu** [minimalist-components/mn-button#1]
Hi, why not implement markdown tags for this component?
Such as: `<mn-button type="submit" class="action raised"` ... and`<mn-a />` or `<mn-link />`

**Comment by darlanmendonca** [comment]
HI @irineu, thanks for suggestion :D

let me understand, by markdown for this component, you want say what?
looking your code, I understand use this component using a tag `mn-button` instead class `.mn-button` is that?

**Comment by irineu** [comment]
Sorry, updated the post, i've forgot the comment markdown scape characters

**Comment by darlanmendonca** [comment]
yep, but the suggestion is that right? use the custom tags `<mn-buttton>` or `<mn-link>`?

**Comment by irineu** [comment]
Yes, that is the idea... I've seen others components such as mn-select, mn-input, mn-radio, mn-textarea... And the majority of these components have a custom html markdown (i.e `<mn-* />`), i think of this a nice way to identify if a button or a link is a mn component or a primitive html component. I've checked out the button features and for now it looks like only css customisation, but it could have in the future some useful features like toggle, enable after complete/validate the form, custom events and so on using element tags in your component. What you think?

**Comment by darlanmendonca** [comment]
I like, the advantages of a components based on tag, is

- more readable
- easy to add features without breaking a previous implementations (because the entire markup is setted by a class js instead manually in markup

you idea to enable after complete/validate looks awesome!

at moment, I dont implement this component as a tag, because I use the api customElements to to a custom tag, like

```js
class MnButton extends HTMLElement {
  constructor(self) {
    self = super(self)
    // here goes things to do on create a mn-button
    // e.g. this.method1()
    return self
  }

  method1() {

  }

  method2() {

  }
}

// here I register the tag mn-button with this class as prototype
window.customElements.define('mn-button', MnButton)
```

note, I extend HTMLElement, and has ways to extends HTMLButtonElement, because button has specific behavior, 

e.g., inside a form, the button is by default a submit, out of a form, is just a button.
now, the tag `<a>` has specific behaviors to, like open the url in a new tab, when user click with the key ctrl (or command in macOS) to open the url in a new tab

this things, are important, I need extends this, and not reimplement all again :D, to dont generate a heavy component

actually, the css of this component have only 1kb (500b if gziped)
with an addiction of js class to work, we need maintain the low size :D

**Comment by darlanmendonca** [comment]
I will create a branch, to add the class, and begin to test the extends of HTMLButtonElement

And I mention that here, to us

In docs of https://github.com/WebReflection/document-register-element, library to fallback native api customElements, I found mention to how extends, but I need found a way to do right, I tried ealy and dont work correctly.

**Comment by darlanmendonca** [comment]
An idea, is make this component using only the tag `<mn-button>texto</mn-button>`

If dont have href, they extend button
If have attr href, extend  anchor `<a>`

what u think?

**Comment by darlanmendonca** [comment]
the branch https://github.com/minimalist-components/mn-button/tree/custom-tag
